lib/util.js

```diff
@@ -174,40 +322,62 @@ const updateDBForCRXFile = (endpoint, region, crxFile, componentId, contentHash)
 }
 
 // Update incrementVersion if this code is updated
-const getPreviousVersion = (version) => {
+const getPreviousVersion = (version, diff = 1) => {
   if (version === FirstVersion) {
     // This is the first version
     return FirstVersion
   }
   const versionParts = version.split('.')
-  versionParts[versionParts.length - 1]--
+  versionParts[versionParts.length - 1] -= diff
   return versionParts.join('.')
 }
 
-const uploadExtension = async (name, id, version, crxFile) => {
-  const S3_EXTENSIONS_BUCKET = process.env.S3_EXTENSIONS_BUCKET || 'brave-core-ext'
-  const componentFilename = `extension_${version.replace(/\./g, '_')}.crx`
-  const componentName = `${name.replace(/\s/g, '-')}`
-  const previousComponentFilename = `extension_${getPreviousVersion(version).replace(/\./g, '_')}.crx`
-  // See tag restrictions here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions
-  const componentNameTag = `${componentName.replace(/[^a-zA-Z0-9+\-=._:/@]+/g, '')}`
-  const latestVersionTagKey = 'version'
-  const latestVersionTagValue = `${componentNameTag}/latest`
+const uploadToS3 = async (file, bucket, key, contentType) => {
   const putObjectParams = {
-    Body: await readFile(crxFile),
-    Bucket: S3_EXTENSIONS_BUCKET,
-    Key: `release/${id}/${componentFilename}`,
+    Body: await readFile(file),
+    Bucket: bucket,
+    Key: key,
     GrantFullControl: process.env.S3_CANONICAL_ID,
     GrantRead: process.env.CLOUDFRONT_CANONICAL_ID,
-    ContentType: 'application/x-chrome-extension'
+    ContentType: contentType
   }
   await s3Client.send(new PutObjectCommand(putObjectParams))
     .catch(err => {
-      console.error(`Upload failed for s3://${S3_EXTENSIONS_BUCKET}/${id}/${componentFilename}`)
+      console.error(`Upload failed for s3://${bucket}/${key}`)
       console.error('Unable to upload:', err.stack, 'Do you have ~/.aws/credentials filled out?')
       throw new Error('Failed to upload extension to S3')
     })
-  console.log(`Uploaded component to s3://${S3_EXTENSIONS_BUCKET}/${id}/${componentFilename}`)
+}
+
+const uploadExtension = async (name, id, version, hash, crxFile) => {
+  const componentFilename = `extension_${version.replace(/\./g, '_')}.crx`
+  const componentName = `${name.replace(/\s/g, '-')}`
+  const previousComponentFilename = `extension_${getPreviousVersion(version).replace(/\./g, '_')}.crx`
+  // See tag restrictions here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions
+  const componentNameTag = `${componentName.replace(/[^a-zA-Z0-9+\-=._:/@]+/g, '')}`
+  const latestVersionTagKey = 'version'
+  const latestVersionTagValue = `${componentNameTag}/latest`
+  const patchDir = `build/patches/${id}/${hash}`
```

Comment:
can we use `version` instead of `hash`? That will make it significantly easier to audit/cleanup old patches in the future, if needed